### DISCUSSION TITLE
How to start and stop scheduled jobs?

### DISCUSSION THREAD

**Discussion by dthomas-sensonix** [go-co-op/gocron#422] [Q&A] [answered]
I have a system where I would like to schedule one job to run at a fixed interval of time.  I would also like to be able to start and stop that job.  Currently I am starting the scheduler with `scheduler.StartAsync()`.

I start the job with `scheduler.Every(config.Config.Interval.Collection).Seconds().Do(collectData)`.

I stop the job with `scheduler.Remove(collectData)`.

While this works, I can encounter scenarios where two instances of the job will run at the same time.  This happens if I stop the job while it is running and restart it immediately.  It appears that the scheduler views the subsequent call to `Do(collectData)` as a new job and does not honor any SingletonMode controls.  This makes sense, but I would like a way to wait for the removed jobs to finish before I allow the user to start new jobs.  I don't want to cancel any currently running jobs.  Is there a way to do that?

I tried calling `scheduler.Stop()` to stop the job.  While that worked, when I called `scheduler.StartAsync()` again the code generated a panic.  Is there a way to clean up after a `Stop()` or reinitialize the scheduler such than it can be restarted?  What is the preferred method here?

**Comment by JohnRoesler** [comment, marked as answer]
Hi @dthomas-sensonix i missed this discussion from you. The issue with stopping and restarting the scheduler has been fixed. Please pull the latest. Stop waits for all jobs to finish before returning.

**Reply by rsaxena05** [reply]
which version was it fixed on ?

**Reply by JohnRoesler** [reply]
https://github.com/go-co-op/gocron/releases/tag/v1.22.2
